tests/system/presenter/test_search_deleted_models.py

```diff
@@ -0,0 +1,140 @@
+from typing import Any
+
+from .base import BasePresenterTestCase
+
+
+class TestSearchDeletedModels(BasePresenterTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.users = {
+            i: {
+                "id": i,
+                "first_name": f"first{i}",
+                "last_name": f"last{i}",
+                "username": f"user{i}",
+            }
+            for i in range(2, 6)
+        }
+        self.motions = {
+            i: {
+                "id": i,
+                "title": f"title{i}",
+            }
+            for i in range(1, 5)
+        }
+        self.motions[1]["number"] = "42"
+        self.assignments = {
+            i: {
+                "id": i,
+                "title": f"title{i}",
+            }
+            for i in range(1, 5)
+        }
+        user_group = {
+            "group_$1_ids": [1],
+        }
+        meeting_id = {
+            "meeting_id": 1,
+        }
+        # add not deleted models
+        self.set_models(
+            {
+                "meeting/1": {
+                    "user_ids": [5],
+                    "motion_ids": [4],
+                    "assignment_ids": [4],
+                    "group_ids": [1],
+                },
+                "group/1": {
+                    "meeting_id": 1,
+                },
+                "user/5": self.users[5] | user_group,
+                "motion/4": self.motions[4] | meeting_id,
+                "assignment/4": self.assignments[4] | meeting_id,
+            }
+        )
+        # add deleted models
+        self.set_models(
+            {
+                "user/2": self.users[2] | user_group,
+                "user/3": self.users[3] | user_group,
+                "user/4": self.users[4],
+                "motion/1": self.motions[1] | meeting_id,
+                "motion/2": self.motions[2] | meeting_id,
+                "motion/3": self.motions[3],
+                "assignment/1": self.assignments[1] | meeting_id,
+                "assignment/2": self.assignments[2] | meeting_id,
+                "assignment/3": self.assignments[3],
+            },
+            True,
+        )
+
+    def search_deleted_models(
+        self,
+        collection: str,
+        filter_string: str,
+        meeting_id: int = 1,
+        status_code: int = 200,
+    ) -> Any:
+        actual_status_code, data = self.request(
+            "search_deleted_models",
+            {
+                "collection": collection,
+                "filter_string": filter_string,
+                "meeting_id": meeting_id,
+            },
+        )
+        self.assertEqual(actual_status_code, status_code)
+        return data
+
+    def test_search_users_first_name(self) -> None:
```

Comment:
Please ad tests for the error cases:
- searching without meeting_id ( required or not)
- searching with other collections/not existing connection

Reply:
What is the real use case for this? Searching for mistakenly deleted users. This is what the presenter can do and don't need the autoupdate-service for this.
With the idea that we don't want the presenter anymore, we should only implement this here, if this an urgently needed feature. After 3 month I'm in doubt about this and think we should close this issue.

Reply:
> What is the real use case for this? 

Not only mistakenly deleted users, but any model which was deleted (by accident or not) somewhere down the line and one wants to access any history data about it. If I don't know the fqid of the model im looking for (which I most certainly do not, as a normal user), the only way to search for the information I want is by accessing a random time in the history of another model which still exists, which lays before the time the model I'm searching for was deleted. Then I'm able to write down the fqid and access the history data of it. This is a valid and imported use case which was discussed with @emanuelschuetze when this PR was created.

> This is what the presenter can do and don't need the autoupdate-service for this.

Currently, the client access the history information via the autoupdate route `history_information`. This lead to https://github.com/OpenSlides/openslides-autoupdate-service/issues/586 being created, since the route is broken for deleted models. Another way would be to use the backend presenter `get_history_information` (which was created by mistake, I think, and in parallel to the AU route), which would need an update of the client and probably changes in the backend presenter, but would be possible. It's important to differ between the two presenters `get_history_information` (already exists) and `search_deleted_models` (implemented in this PR), since they do completely different things.

> With the idea that we don't want the presenter anymore

Before just abandoning presenters, we need a replacement system for it, which I don't see yet.

> After 3 month I'm in doubt about this and think we should close this issue.

It was a feature request 3 months ago. If @emanuelschuetze now decides that we don't need it anymore, we can close it, otherwise I think since https://github.com/OpenSlides/openslides-autoupdate-service/issues/586 was closed without intention to change the AU the way to go would be to change the client to use the `get_history_information` presenter so this can be finalized and merged. This would also supposedly make @ostcar happy since the `history_information` route could be deleted from the AU ;)